pylint/extensions/typing.py

```diff
@@ -467,22 +473,25 @@ def _check_broken_noreturn(self, node: nodes.Name | nodes.Attribute) -> None:
             # NoReturn not part of a Union or Callable type
             return
 
-        if (
-            in_type_checking_block(node)
-            or is_postponed_evaluation_enabled(node)
+        if in_type_checking_block(node) or (
+            is_postponed_evaluation_enabled(node)
             and is_node_in_type_annotation_context(node)
         ):
             return
 
         for inferred in node.infer():
             # To deal with typing_extensions, don't use safe_infer
             if (
-                isinstance(inferred, (nodes.FunctionDef, nodes.ClassDef))
-                and inferred.qname() in TYPING_NORETURN
+                (
+                    isinstance(inferred, (nodes.FunctionDef, nodes.ClassDef))
+                    and inferred.qname() in TYPING_NORETURN
+                )
                 # In Python 3.7 - 3.8, NoReturn is alias of '_SpecialForm'
-                or isinstance(inferred, astroid.bases.BaseInstance)
-                and isinstance(inferred._proxied, nodes.ClassDef)
-                and inferred._proxied.qname() == "typing._SpecialForm"
+                or (
+                    isinstance(inferred, astroid.bases.BaseInstance)
+                    and isinstance(inferred._proxied, nodes.ClassDef)
+                    and inferred._proxied.qname() == "typing._SpecialForm"
+                )
             ):
```

Comment:
I know I'm a bit late on this one, however I'm not sure this is really an improvement. Took me a while to figure out what was causing this change -> [`RUF021`](https://docs.astral.sh/ruff/rules/parenthesize-chained-operators/).

It might make sense for regular code bases but I don't think that fully applies to pylint / astroid. We often use convoluted if statements with very specific formatting and even comments to a) reduce the indentation and b) keep it readable. The linting rule doesn't help with that IMO.

What do you guys think?

/CC @Pierre-Sassoulas @jacobtylerwalls

Reply:
I liked this change because it's clearer imo (knowing that 'and' is prioritary should be an automatism but it's not for me and it helps me understand the code faster). What's not to like ? The conditional taking more space/line ?

Reply:
> The conditional taking more space/line ?

Yeah. I'm just not a huge fan of tools forcing me to write code a certain way just because it thinks it's better.
Anyway, probably not big deal in the grand scheme of things. _Will just not use it for my personal projects then._